### PR TITLE
Temporary fix for sl fallback + propagateHelixToPlaneMPlex call

### PR DESCRIPTION
**Pull request by leonardogiannini**
This is just to fix the usage of the straight line fall back in prop to plane 

the minimal check to use the fallback is in the comment, however so far it gives me worse results on the central low pT muons
image

vs no check at all

image
 
so so more investigation is needed

The other fixes the call of propagateHelixToPlaneMPlex to what is in the template and what I found in the debug code.
The differences in the physics are barely noticeable

### CODE REVIEW

File: RecoTracker/MkFitCore/src/PropagationMPlexPlane.cc [hunk 1]
```diff
@@ -489,7 +490,7 @@ namespace {
                   << " std::isfinite(s[n])=" << std::isfinite(s[n]) << " std::isnormal(s[n])=" << std::isnormal(s[n])
                   << std::endl;
 #endif
-      if ((std::abs(sl[n]) > std::abs(s[n])) || std::isnormal(s[n]) == false)
+      if (0)  //(mkfit::isFinite(s[n])==false && mkfit::isFinite(sl[n])) // needs more checks because it looks worse
```
Comment (slava77) [inline RecoTracker/MkFitCore/src/PropagationMPlexPlane.cc:493]: some kind of ifdef here would make more sense

**Comment by leonardogiannini** [comment]
as expected phase 1 is not affected by changes in P2P
http://uaf-10.t2.ucsd.edu/~legianni/plots_validateP2P_sl/plots_phase1/

for the phase2 I see that mkFit initial step built tracks are almost unaffected
image

but there is some loss at pT>10 GeV for the loose and tight selection
image

where I'm comparing the phase 2 CKF (blue) to the mkFit default (red) and PR (black)

plots can be found here http://uaf-10.t2.ucsd.edu/~legianni/plots_validateP2P_sl/plots_phase2_WF.0vs702

so we should investigate when the straight line is necessary

**Comment by slava77** [comment]
> but there is some loss at pT>10 GeV for the loose and tight selection

... it's not really until pt>40 GeV.

I don't see significant differences in byOriginalAlgo plots.
It looks like some track properties degrade compared to other iterations that reconstruct the same track.
I don't see a significant increase downstream in highPtTriplet, but there is a significant increase in the muon InOut iteration.

It looks like the changes in pt>40 are for the muons: running on a muon gun sample would help, since it's easy to debug.

The list of plots is fairly short: is it an issue of not having `trackingOnly` components in the workflow or just a matter of running the comparison script?

**Comment by leonardogiannini** [comment]
with the new commit no change is visible in ttbar phase 2 (even all tracks vs original algo in initial step)
see http://uaf-10.t2.ucsd.edu/~legianni/plots_validateP2P_sl/plots_tt/

it looks like there is an increase in n. of hits in the muon sample at high pT (e.g. http://uaf-10.t2.ucsd.edu/~legianni/plots_validateP2P_sl/plots_3_mu_pt1to1000_eta0p0to0p4)

In the usage of the p2p in the phase-1 refit we see a worsening in resolution when the straight line is used, but otherwise the tracks would be lost (and only the good ones would be in the resolution plot), so we should use the fallback and possibly reduce the s calculation frequency of NANs

**Pull request merged**